Sources/Commands/SwiftTestCommand.swift

```diff
@@ -774,13 +792,25 @@ extension SwiftTestCommand {
                     )
 
                     // Finally, run the tests.
+                    var output = [String]()
                     let result = runner.test(outputHandler: {
-                        // command's result output goes on stdout
-                        // ie "swift test" should output to stdout
-                        print($0, terminator: "")
+                        output.append($0)
                     })
                     if result == .failure {
                         swiftCommandState.executionStatus = .failure
+                        // If the runner reports failure do a check to ensure
+                        // all the binaries are present on the file system.
+                        for path in testProducts.map(\.binaryPath) {
+                            if !swiftCommandState.fileSystem.exists(path) {
+                                throw FileSystemError(.noEntry, path)
```

Comment:
Errors thrown from here still don't flush accumulated `output`, which IIUC is a change from how it previously worked?

Reply:
I've reverted this. In the failure case nothing was printed, but if there is more than one swift-testing testProduct then the successful ones should still print.